helix-core/src/main/java/org/apache/helix/participant/DistClusterControllerStateModel.java

```diff
@@ -40,6 +40,9 @@ public class DistClusterControllerStateModel extends AbstractHelixLeaderStandbyS
   protected Optional<HelixManager> _controllerOpt = Optional.empty();
```

Comment:
Given we are reinstantiating this in multiple threads, its recommended to make this volatile to prevent processor using the stale reference from the cache.